inc/init.php

```diff
@@ -133,6 +133,10 @@ function delta_time($start=0) {
     define('DOKU_TPLINC', DOKU_INC.'lib/tpl/'.$conf['template'].'/');
 }
 
+// Self-help URL for DokuWiki Setup Error page
+define('DOKU_SETUP_SELFHELP_URL', "https://www.dokuwiki.org/install:permissions");
+define('DOKU_SETUP_SELFHELP_LINK',"<hr />You may wish to:<ul><li>Run the <a href=\"install.php\">DokuWiki installer</a></li><li>Review <a href=\"$self_help_url\">https://www.dokuwiki.org/install:permissions</a></li></ul>");
```

Comment:
I guess only one constant `DOKU_SETUP_SELFHELP_LINK` should be enough. Also I never see where `$self_help_url` is (which means it will never work).

Reply:
Thank you @phy25. I used two constants so that the URL need only be explicitly written out once. If you strongly prefer that there be a single constant with the URL written out twice, I can do that of course. The `$self_help_url` is my mistake and must be corrected.

Reply:
I am still not sure where `DOKU_SETUP_SELFHELP_URL` is used (more than one)... My thought is eventually `DOKU_SETUP_SELFHELP_LINK` is used.

Reply:
It isn't - that's my mistake - but the intention is that the code should contain both a link to the URL (the `href` attribute) and the link should also display the URL as the text of the link, something like this pseudocode:

`<a href="DOKU_SETUP_SELFHELP_URL">DOKUWIKI_SETUP_SELFHELP_URL<a>`

The intention is to avoid writing out the actual URL twice, since if it ever changes we would then have to change it in two places. Thank you for taking the time to review my PR. I apologize for the confusion.

Reply:
Yeah that makes sense. It's fine to make two constants then.

By the way thank you for following up - everyone is busy (and that's why I left this PR email in my inbox without doing the actual review for almost a month).

Reply:
OK my initial thought when writing my last comment is that `DOKU_SETUP_SELFHELP_URL` can be changed  by third-party developers or the environment without touching `DOKU_SETUP_SELFHELP_LINK`; but it seems that it will not work (You have to replace strings for `DOKU_SETUP_SELFHELP_URL` when reading `DOKU_SETUP_SELFHELP_LINK` which makes extra work). Also I don't think it is valuable just to avoid duplicate (= 2) occurrences of the URL, just in the link string. So, I prefer keeping only one `DOKU_SETUP_SELFHELP_URL` constant here.